torch_object_detection_segmenter.py

```diff
@@ -1,31 +1,194 @@
+                assert (w_beg is not None and h_beg is not None)
+                assert (0 <= w_beg <= (img_w - target_w)), f'left must be within [0, {img_w - target_w}]: {w_beg}'
+                assert (0 <= h_beg <= (img_h - target_h)), f'top must be within [0, {img_h - target_h}]: {h_beg}'
+            else:
+                raise ValueError(f'unknown input how: {how}')
+            if not isinstance(w_beg, int):
+                raise ValueError(f'left must be int number between 0 and {img_w}: {left}')
+            if not isinstance(h_beg, int):
+                raise ValueError(f'top must be int number between 0 and {img_h}: {top}')
+            w_end = w_beg + target_w
+            h_end = h_beg + target_h
+            img = img.crop((w_beg, h_beg, w_end, h_end))
+            return img, h_beg, w_beg
+
+        blob = docs.get_attributes('blob')
+        batch = np.copy(blob[0]) # (2, 681, 1264, 3) with imgs/cars.jpg
+        # "Ensure the color channel axis is the default axis." i.e. c comes first
+        # e.g. (h,w,c) -> (c,h,w) / (b,h,w,c) -> (b,c,h,w)
+        batch = _move_channel_axis(batch, self.channel_axis, self._default_channel_axis + 1) # take batching into account
+
+        batched_predictions = self._predict(batch)
+
+        result = []
+        for image, predictions in zip(batch, batched_predictions):
+            bboxes = predictions['boxes'].detach()
+            scores = predictions['scores'].detach()
+            labels = predictions['labels']
+            if self.on_gpu:
+                bboxes = bboxes.cpu()
+                scores = scores.cpu()
+                labels = labels.cpu()
+            img = _load_image(image * 255, self._default_channel_axis)
+
+            batched = []
+            for bbox, score, label in zip(bboxes.numpy(), scores.numpy(), labels.numpy()):
+                if score >= self.confidence_threshold:
+                    x0, y0, x1, y1 = bbox
+                    # note that tensors are [H, W] while PIL Images are [W, H]
+                    top, left = int(y0), int(x0)
+                    # target size must be (h, w)
+                    target_size = (int(y1) - int(y0), int(x1) - int(x0))
+                    # at this point, raw_img has the channel axis at the default tensor one
+                    _img, top, left = _crop_image(img, target_size=target_size, top=top, left=left, how='precise')
+                    _img = _move_channel_axis(np.asarray(_img).astype('float32'), -1, self.channel_axis)
+                    label_name = self.label_name_map[label]
+                    logging.debug(
+                        f'detected {label_name} with confidence {score} at position {(top, left)} and size {target_size}')
+                    batched.append(
+                        dict(offset=0, weight=1., blob=_img,
+                             location=(top, left), tags={'label': label_name}))
+
+            result.append(batched)
+
+        return result
```

Comment:
this is wroong. U do not want to return anything, u want to add `chunks` to the `Document`

Reply:
changed, but technically this old style could work ?